### PR TITLE
Repaired and extended probe supervision

### PR DESCRIPTION
**Pull request by uwiger**
- Delete of failed probe didn't work
- Restart parameterization support in exometer_probe (see edoc)
- Disabled probes (e.g. histograms) are now terminated
- When re-enabled, probes are re-instantitated
- If new entries include aliases, these are checked in advance
- Support for deleting all cached datapoints for a given entry
- Fix exometer:start(), which had become broken
- Make sure eunit tests succeed
- Add ct.config with lager log config
- Fix breakage in CT test case init and cleanup
- Add test suite for supervision functionality

### CODE REVIEW

**Comment by tolbrino** [comment]
kudos for adding test coverage too. The xref errors should be easy fixes. I will dig through this PR tomorrow.

**Comment by uwiger** [comment]
To avoid merge conflicts, I don't re-generate the edoc in PRs. You'll have to read the edoc src.
https://github.com/Feuerlabs/exometer_core/blob/uw-probe-restart/src/exometer_probe.erl#L379

There are also some eunit tests (should be auto-run in make ci, perhaps?) in [exometer_probe.erl](https://github.com/Feuerlabs/exometer_core/blob/uw-probe-restart/src/exometer_probe.erl#L801).

File: src/exometer_admin.erl [hunk 1]
```diff
@@ -261,7 +265,7 @@ start_link() ->
 init(_) ->
     {ok, #st{}}.
 
-handle_call({new_entry, Name, Type, Opts, AllowExisting}, _From, S) ->
+handle_call({new_entry, Name, Type, Opts, AllowExisting} = _Req, _From, S) ->
```
Comment (tolbrino) [inline src/exometer_admin.erl]: Feel weird prefixing the variable and still using in within the function body.

**Comment by tolbrino** [comment]
re Eunit: I'd rather stick to one testing tool, move the tests to CT and use the ifdef switch to export the functions being tested.

re CT tests: lots of noise through a lot of logging, but the tests look good.

There are a few syntactic misalignments, but I won't nitpick on that considering the size of the change. I think they will be aligned once someone touches the code again.

LGTM

side note: Going through the docs again I noticed that they are becoming fairly complex. Maybe a edown to Github Pages converter would make sense to give the docs a place to stay?

**Comment by uwiger** [comment]
There's definitely a point to running everything from CT. What I like with the eunit tests is that they (sometimes) help clarify the code, in the same module.

Am I to interpret your comment as an Ok to merge? :)

**Comment by tolbrino** [comment]
yes go ahead

Regarding eunit: I see your point and am not hard set on CT only. But as you said eunit should then be executed as part of every testrun.

**Comment by uwiger** [comment]
Have done a manual merge, after I merged the report_logger branch into master. But when running `make ci`, I found a dialyzer issue with the report_logger, indicating an actual bug. Not sure why it didn't show up before. Working on it.

**Comment by uwiger** [comment]
That is, manual merge of the new master into this branch.

**Comment by tolbrino** [comment]
Which OTP version were you using? I didn't get any suspicious dialyzer output.

**Comment by uwiger** [comment]
Actually, it shows up in the Travis logs (around line 425) without actually causing a fail (suspicious in itself).

**Pull request merged**